### PR TITLE
[google_maps_flutter] adds support for custom icon from a byte array (PNG)

### PR DESCRIPTION
**Pull request by miguelpruivo**
There is already a way to add an asset (from the asset file list) as a custom icon.
This PR will add a `BitmapDescritpor.fromBytes(Uint8List)` that receives an array of bytes from a Bitmap image. This will then call the already available Maps API that uses a custom `Bitmap` as an override for the Marker icon.

### CODE REVIEW

**Comment by miguelpruivo** [comment]
I signed it!

**Comment by miguelpruivo** [comment]
I signed it!

**Comment by amirh** [comment]
This is great, thanks! We would like to wait until we have maps implemented on both iOS and Android before actually further augmenting the API, because we are worried that they'll get out of sync otherwise. Would you be ok with resubmitting this with iOS support in a few weeks once we have the iOS work more fleshed out?

**Comment by collinjackson** [comment]
Can you update this to add iOS support?

**Comment by miguelpruivo** [comment]
@collinjackson definitely. I might take a look into it yet this week.

**Comment by miguelpruivo** [comment]
@collinjackson done.

**Comment by miguelpruivo** [comment]
It’s ready, waiting to be merged on master @awazgyawali.

**Comment by kjeremy** [comment]
@miguelpruivo Why do all of these PRs take weeks to merge?

**Comment by miguelpruivo** [comment]
@kjeremy maybe because the team hadn’t time yet to merge it. I’ve been also waiting. In the meantime I’ll use a forked branch for my needs.

**Comment by aytunch** [comment]
Instead of bitmaps, why can't we use custom widgets as markers? Is it impossible to implement?

**Comment by miguelpruivo** [comment]
@aytunch although possible, it’s not the raw data and the plugin itself would have to convert to bmp before calling the underlying platforms. Also, you may want to use a bitmap or so that isn’t actually from a widget.

**Comment by miguelpruivo** [comment]
@zoechi any news on this? Currently I have a production project that relies on this. Would be great have it merged.

Rebased today to match upstream changes.

**Comment by m7amdElberawy** [comment]
How to use this work around, I try to Implement this code but not acceptable while debugging...

**Comment by iskakaushik** [comment]
@miguelpruivo , I can drive this PR to completion if you do not have the cycles/time to do it. Please allow changes to your branch. Follow the instructions at: https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork

**Comment by TuttaFM** [comment]
In what version will be this change in Bitmap Descriptor. I'm using version 0.5.2  of google map . this change in bitmap.dart isn't exist.

File: packages/google_maps_flutter/lib/google_maps_flutter.dart [hunk 1]
```diff
@@ -5,6 +5,7 @@
 library google_maps_flutter;
 
 import 'dart:async';
+import 'dart:typed_data';
```
[the remark this answers is not in the document]
Comment (iskakaushik) [reply packages/google_maps_flutter/lib/google_maps_flutter.dart]: ignore this comment

**Pull request merged**